### PR TITLE
Remove WAN DDNS

### PR DESCRIPTION
**Pull request by sparky8251**
It's odd that JF still had code lying around for generating a self signed cert. Currently, it does not do this so this code has been removed.

JF also appears to have functions in place to modify provided certs? Warrants deeper investigation. JF should not be attempting modifications of any certs under any circumstance.

### CODE REVIEW

**Comment by sparky8251** [comment]
Should be merged with https://github.com/jellyfin/jellyfin-web/pull/472

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 1]
```diff
@@ -1198,25 +1198,11 @@ private IEnumerable<string> GetUrlPrefixes()
 
         private CertificateInfo GetCertificateInfo(bool generateCertificate)
```
Comment (cvium) [inline Emby.Server.Implementations/ApplicationHost.cs]: I'm not 100% on the syntax
```suggestion
        private CertificateInfo GetCertificateInfo() => new CertificateInfo
        {
            Path = ServerConfigurationManager.Configuration.CertificatePath,
            Password = ServerConfigurationManager.Configuration.CertificatePassword
        };
```
cc @Bond-009

**Comment by anthonylavado** [comment]
Wrong issue # on my part, disregard

File: Emby.Server.Implementations/ApplicationHost.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/ApplicationHost.cs]: I'm not sure either, I'd say leave it like this, we'll find out if the analyzers don't like it.

**Comment by anthonylavado** [comment]
@tolriq Will this impact Yatse at all?

**Comment by Tolriq** [comment]
Answered on forums but not sure to understand why you are removing all those, many users use Jellyfin to access their media out of home. WAN IP/certificates are a must.

**Comment by sparky8251** [comment]
The WAN IP in `/system/info` and `/system/info/public` is what is being removed.

This change does not impact remote access as all clients (at least ours...) currently use the IP provided during initial connection. Currently, it is looking like this was used for Emby Conenct (which we don't have and have 0 plans to implement).

I mean, you have to have the IP of the server to even get this API response, and if you are on the WAN you already have the WAN IP when you connect. This is only if for some reason a client on the LAN needs the WAN address of the server, which we have no instances of on our end.

**Comment by Tolriq** [comment]
Well I do :) 
99% of users do configure their instance on the LAN via the auto detection protocol.
Then Yatse have access to both IP and can switch when out of LAN automatically.

This is currently not done for Jellyfin and only for Plex as they properly handle the whole chain including proper certificates but it was on the TODO. I guess I can remove that line, but if the user can enter his external DNS name in the server config and it's exposed to the API this is good for advanced / modern Jellyfin clients that offers advanced features.

**Comment by sparky8251** [comment]
And this is why we ask :)

On our end, currently the primary driving force of removing this WAN IP is that it requires we have some infrastructure or 3rd party to handle telling the server what its WAN IP is. We as a project want none of this shenaniganery. We could add it in as a manually filled out field allowing IPs or DNS addresses...

I mean... Allowing a manually entered domain would probably retain some of the functionality you want. I have no objections to it and I doubt anyone else will. I'm not a fan of allowing a manually entered IP however. DDNS exists to tackle changing WAN IPs, JF cannot implement and IP form of DDNS without a 3rd party, which we don't want. Hence all this to begin with.

**Comment by sparky8251** [comment]
Just as a point of note, if someone uses a reverse proxy this auto-connect of yours wouldn't work. Reverse proxies route based on the domain name in the HTTP header, not IPs.

So a direct IP connection will break in the instance of a reverse proxy.

**Comment by oddstr13** [comment]
I would prefer a admin configured *URL* rather than domain.

Jellyfin may be at any subpath based on reverse proxy configuration.

**Comment by Tolriq** [comment]
https://www.ipify.org/ :)

About IP depending on countries many have fixed IP + all those who rent a server and share between friends.

Anyway since I currently have not yet implemented this for Jellyfin this is no big deal if you remove, but if you add something in the future that is more robust / support dns names please ping so I can add that back to the TODO list. My internals are ready for that, it was delayed due to massive usage of self signed certs for Emby / JellyFin that I did not want to blindly accept, but without triggering the user into options most would not understand.

For your edit, for those advanced use case users are able to correctly configure Yatse ;) And Yatse support also subdirectory proxying configuration that most prefers over domains are simpler to configure and limit certificates management.

**Comment by sparky8251** [comment]
For what its worth, we already have a field that's filled with an external domain.

It's entire reason for being is dumb (its used to make self signed certs with a valid domain, but all clients will bitch about the cert being self signed regardless of a valid domain or not, plus most clients wont let you inspect the cert to even verify the domain so... yeah).

I was planning on removing it, but it might be better to repurpose it. However, if we did so I would only put it on the authenticated `/system/info` endpoint and not the public one. URL structure being exposed is far more sensitive than an IP after all.

**Comment by Tolriq** [comment]
Yes only authenticated is OK users either use auto detection that only works on LAN or is able to properly configure the external from the start.
And yes self signed certificates are dumb as users do not understand and blindly accept the prompt to validate without looking :(

**Comment by davilla** [comment]
Removing WanAddress is fine BUT client apps really need a bit more notice to resolve issues (if any). In Emby, it is a required field and  now it would be optional. So we would have to alter code to account for that behavior.

https://github.com/MrMC/mrmc/search?q=WanAddress&unscoped_q=WanAddress

Altering code is not an issue but getting it released can be. For us (MrMC) as we are in the various App stores (Apple, Amazon and GooglePlay), it takes several weeks (or more) to get a new version released. For example, we just pushed out MrMC v3.9.2. It cleared Amazon, GooglePlay and Apple/tvOS but Apple/iOS is being held up because of stupid issues with novice reviewers :) Going on week two now of trying to get approval. This is the second time in a row this has happened for Apple/iOS (MrMC v9.3.1) after flying through reviews for several years of releases.

I suggest sticking in the loopback address (127.0.0.1), access will fail but existing code usage should not crash. For a few months, anyone that is using it will be forced to take notice :). Then nuke it out. MrMC v3.9.3 will have treat WanAddress as optional.

**Comment by sparky8251** [comment]
So, I just realized that this PR is the wrong one to be discussing this on... This is a removal for something other than the WAN IP that is so useless no one even noticed this PR was for something other than what is being discussed XD

Anyways, I think @anthonylavado confirmed that Apple TV worked fine with the WAN removal PR in place with MrMC.

Do you share code for platforms @davilla ? If so, we are probably fine.

**Comment by davilla** [comment]
Nice. yes the code is shared across the platforms we support.

**Comment by anthonylavado** [comment]
> Nice. yes the code is shared across the platforms we support.

My bad for using this PR. Yes, @thornbill confirmed the currently shipping MrMC worked on iOS, and I confirmed tvOS. (Thanks again @davilla!)

**Pull request merged**